src/ui.mjs

```diff
@@ -1,18 +1,39 @@
-const input = document.querySelector('#input');
-const calculate = document.querySelector('#calculate');
-const source = document.querySelector('#source');
-const ast = document.querySelector('#ast');
+import setupTabs from './tabs.mjs';
 
-function astListener(text) {
-  ast.textContent = text;
+function initTabs() {
```

Comment:
Each of these could be a module of their own, but that's just a personal preference.